src/main/java/org/folio/consortia/config/kafka/KafkaRecordInterceptor.java

```diff
@@ -0,0 +1,59 @@
+package org.folio.consortia.config.kafka;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+import org.folio.spring.DefaultFolioExecutionContext;
+import org.folio.spring.FolioModuleMetadata;
+import org.folio.spring.integration.XOkapiHeaders;
+import org.folio.spring.scope.FolioExecutionScopeExecutionContextManager;
+import org.springframework.kafka.listener.RecordInterceptor;
+import org.springframework.stereotype.Component;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@Component
+@RequiredArgsConstructor
+public class KafkaRecordInterceptor<V> implements RecordInterceptor<String, V> {
```

Comment:
The same with kafka record interceptor https://github.com/folio-org/mod-data-export-spring/pull/240/files